zigpy/util.py

```diff
@@ -111,12 +111,14 @@ async def retry(func, retry_exceptions, tries=3, delay=0.1):
             return r
         except retry_exceptions:
             if tries <= 1:
+                if async_on_failure is not None:
+                    await async_on_failure
```

Comment:
Wouldn't you want `await async_on_failure(e)` or something? Otherwise, if an exception isn't raised, it seems like this would generate a "coroutine not awaited" warning.

Reply:
yes, You are correct, should just pass the callable and await it here.